src/web/handlers.rs

```diff
@@ -1,65 +1,59 @@
 
 pub const ONE_KB: f64 = 1024.0;
 
-pub fn get_collections(meta: MetaRequest) -> FutureResponse<HttpResponse> {
-    Box::new(
-        meta.db
-            .get_collection_timestamps(meta.user_id)
-            .map_err(From::from)
-            .map(|result| {
-                HttpResponse::build(StatusCode::OK)
-                    .header("X-Weave-Records", result.len().to_string())
-                    .json(result)
-            }),
-    )
+pub fn get_collections(meta: MetaRequest) -> impl Future<Item = HttpResponse, Error = Error> {
+    meta.db
+        .get_collection_timestamps(meta.user_id)
+        .map_err(From::from)
+        .map(|result| {
+            HttpResponse::build(StatusCode::OK)
+                .header(X_WEAVE_RECORDS, result.len().to_string())
+                .json(result)
+        })
 }
 
-pub fn get_collection_counts(meta: MetaRequest) -> FutureResponse<HttpResponse> {
-    Box::new(
-        meta.db
-            .get_collection_counts(meta.user_id)
-            .map_err(From::from)
-            .map(|result| {
-                HttpResponse::build(StatusCode::OK)
-                    .header("X-Weave-Records", result.len().to_string())
-                    .json(result)
-            }),
-    )
+pub fn get_collection_counts(meta: MetaRequest) -> impl Future<Item = HttpResponse, Error = Error> {
+    meta.db
+        .get_collection_counts(meta.user_id)
+        .map_err(From::from)
+        .map(|result| {
+            HttpResponse::build(StatusCode::OK)
+                .header(X_WEAVE_RECORDS, result.len().to_string())
+                .json(result)
+        })
 }
 
-pub fn get_collection_usage(meta: MetaRequest) -> FutureResponse<HttpResponse> {
-    Box::new(
-        meta.db
-            .get_collection_usage(meta.user_id)
-            .map_err(From::from)
-            .map(|usage| {
-                let usage: HashMap<_, _> = usage
-                    .into_iter()
-                    .map(|(coll, size)| (coll, size as f64 / ONE_KB))
-                    .collect();
-                HttpResponse::build(StatusCode::OK)
-                    .header("X-Weave-Records", usage.len().to_string())
-                    .json(usage)
-            }),
-    )
+pub fn get_collection_usage(meta: MetaRequest) -> impl Future<Item = HttpResponse, Error = Error> {
+    meta.db
+        .get_collection_usage(meta.user_id)
+        .map_err(From::from)
+        .map(|usage| {
+            let usage: HashMap<_, _> = usage
+                .into_iter()
+                .map(|(coll, size)| (coll, size as f64 / ONE_KB))
+                .collect();
+            HttpResponse::build(StatusCode::OK)
+                .header(X_WEAVE_RECORDS, usage.len().to_string())
+                .json(usage)
+        })
 }
 
-pub fn get_quota(meta: MetaRequest) -> FutureResponse<HttpResponse> {
+pub fn get_quota(meta: MetaRequest) -> impl Future<Item = HttpResponse, Error = Error> {
     Box::new(
```

Comment:
I see a lot of these handlers remove the Box::new, but this one doesn't. Is there a reason for that?

Reply:
These don't respond with `HttpResponse::build()`. Might be some argument that we should normalize all of these to do that.

Reply:
Huh, looks like we can drop the boxes, so I will.

Reply:
Yeah that was confusing me that the return type was impl Future and it was returning a Box. I guess Box implements Future by dereferencing itself first?

Reply:
Not quite: the future module implements Future for all Box<F> where F: Future

Reply:
Right, that makes sense. Thanks.